### PR TITLE
migration to new design substation

### PR DESCRIPTION
**Pull request by anistouri**
Signed-off-by: TOURI ANIS <anis-1.touri@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 1]
```diff
@@ -300,7 +298,13 @@ public EquipmentCreationEntity createTwoWindingsTransformerEntity(String id, Str
     }
 
     public EquipmentCreationEntity createSubstationEntity(String id, String name, Country country, Map<String, String> properties) {
-        return new SubstationCreationEntity(id, name, country, properties);
+        return new SubstationCreationEntity(SubstationCreationInfos.builder()
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:301]: You can remove the 'new'

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/SubstationCreationEntity.java [hunk 2]
```diff
@@ -63,5 +70,6 @@ public SubstationCreationInfos toSubstationCreationInfos() {
                 .equipmentName(getEquipmentName())
                 .substationCountry(getCountry()).properties(getProperties());
     }
+
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/creation/SubstationCreationEntity.java:73]: It lacks a
```
@Override
public void cloneWithIdsToNull() {
    super.cloneWithIdsToNull();
    if (this.properties != null) {
        this.properties = new HashMap<>(properties);
    }
}
```
for (forgotten) test for properties in matcher to succeed

Comment (laurgarn) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/SubstationCreationEntity.java:73]: An another mean, as it seems cloneWithIdsToNull is part of the legacy interface we do not want to use anymore.

Comment (laurgarn) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/SubstationCreationEntity.java:81]: I added it myself. The trick around JPA, with this cloneWithIdsToNull will have to be re-evaluated.

**Review by laurgarn — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**